### PR TITLE
Fix incorrect parent/child relationship in inferred span

### PR DESCRIPTION
**Pull request by felixbarny**
## Checklist

- [ ] This is an enhancement of existing features, or a new feature in existing plugins
  - [ ] I have updated [CHANGELOG.asciidoc](https://github.com/elastic/apm-agent-java/blob/main/CHANGELOG.asciidoc)
  - [ ] I have added tests that prove my fix is effective or that my feature works
  - [ ] Added an API method or config option? Document in which version this will be introduced
  - [ ] I have made corresponding changes to the documentation
- [x] This is a bugfix
  - [x] I have updated [CHANGELOG.asciidoc](https://github.com/elastic/apm-agent-java/blob/main/CHANGELOG.asciidoc)
  - [x] I have added tests that would fail without this fix
- [ ] This is a new plugin
  - [ ] I have updated [CHANGELOG.asciidoc](https://github.com/elastic/apm-agent-java/blob/main/CHANGELOG.asciidoc)
  - [ ] My code follows the [style guidelines of this project](https://github.com/elastic/apm-agent-java/blob/main/CONTRIBUTING.md#java-language-formatting-guidelines)
  - [ ] I have made corresponding changes to the documentation
  - [ ] I have added tests that prove my fix is effective or that my feature works
  - [ ] New and existing [**unit** tests](https://github.com/elastic/apm-agent-java/blob/main/CONTRIBUTING.md#testing) pass locally with my changes
  - [ ] I have updated [supported-technologies.asciidoc](https://github.com/elastic/apm-agent-java/blob/main/docs/supported-technologies.asciidoc)
  - [ ] Added an API method or config option? Document in which version this will be introduced
  - [ ] Added an instrumentation plugin? Describe how you made sure that old, non-supported versions are not instrumented by accident.
- [ ] This is something else
  - [ ] I have updated [CHANGELOG.asciidoc](https://github.com/elastic/apm-agent-java/blob/main/CHANGELOG.asciidoc)

### CODE REVIEW

**Comment by eyalkoren** [comment]
@felixbarny I don't remember the details, only that we discussed that there are additional scenarios not fully covered. Since this is getting very old, I propose we decide now whether this change is an improvement on its own and if so - merge. Otherwise let's close it and open an issue so that we can prioritize proper investigation.

**Comment by felixbarny** [comment]
I also don't recall all the details here. But since this does fix an issue in the existing test cases for which we didn't have an assertion, and all existing tests are passing, I think we should merge this.

File: apm-agent-plugins/apm-profiling-plugin/src/test/java/co/elastic/apm/agent/profiler/CallTreeTest.java [hunk 1]
```diff
@@ -611,6 +612,46 @@ void testActivationAfterMethodEnds_RootChangesToB() throws Exception {
...
+            " aaa",
+            "1 1 "
+        }, new Object[][] {
+            {"a",   3},
+        }, new Object[][] {
+            {"a",   3},
+            {"  1", 2},
+        });
+    }     */
+
```
Comment (eyalkoren) [inline apm-agent-plugins/apm-profiling-plugin/src/test/java/co/elastic/apm/agent/profiler/CallTreeTest.java:653]: What about this one? Do you want to keep it as something to fix in the future?

**Review by eyalkoren — APPROVED** [review]

File: apm-agent-plugins/apm-profiling-plugin/src/test/java/co/elastic/apm/agent/profiler/CallTreeTest.java (hunk 1 above)
Comment (felixbarny) [reply apm-agent-plugins/apm-profiling-plugin/src/test/java/co/elastic/apm/agent/profiler/CallTreeTest.java:653]: Yes, I'll leave it in for that reason. Ideally, it should have a comment about what's not working and why but I don't remember the context.

File: apm-agent-plugins/apm-profiling-plugin/src/test/java/co/elastic/apm/agent/profiler/CallTreeTest.java [hunk 2]
```diff
@@ -59,7 +59,6 @@
 import static org.mockito.Mockito.doReturn;
 
 @DisabledOnOs(OS.WINDOWS)
-@DisabledOnAppleSilicon
```
Comment (eyalkoren) [inline apm-agent-plugins/apm-profiling-plugin/src/test/java/co/elastic/apm/agent/profiler/CallTreeTest.java:62]: Ooops!

Comment (eyalkoren) [reply apm-agent-plugins/apm-profiling-plugin/src/test/java/co/elastic/apm/agent/profiler/CallTreeTest.java:62]: @felixbarny I am restoring on a separate PR unless you tell me this is intentional

Comment (felixbarny) [reply apm-agent-plugins/apm-profiling-plugin/src/test/java/co/elastic/apm/agent/profiler/CallTreeTest.java:62]: Whoops, definitely not intentional.

**Pull request merged**